ChaosMod/Effects/db/Misc/MiscVehicleRain.cpp

```diff
@@ -0,0 +1,44 @@
+#include <stdafx.h>
+
+	//Effect by ProfessorBiddle
+
+static void OnTick()
+{
+	Vector3 playerPos = GET_ENTITY_COORDS(PLAYER_PED_ID(), false);
+
+	static DWORD64 lastTick = 0;
+	DWORD64 curTick = GetTickCount64();
+
+	if (curTick > lastTick + 500)
+	{
+		lastTick = curTick;
+
+		Vector3 spawnPos;
+		spawnPos.x = playerPos.x + g_random.GetRandomInt(-100, 100);
+		spawnPos.y = playerPos.y + g_random.GetRandomInt(-100, 100);
+		spawnPos.z = playerPos.z + g_random.GetRandomInt(25, 50);
+
+
+		static std::vector<Hash> vehModels = Memory::GetAllVehModels();
+
+		if (!vehModels.empty())
+		{
+			Vehicle veh = CreatePoolVehicle(vehModels[g_random.GetRandomInt(0, vehModels.size() - 1)], spawnPos.x, spawnPos.y, spawnPos.z, GET_ENTITY_HEADING(PLAYER_PED_ID()));
```

Comment:
Since this effect will spawn a lot of vehicles and the "pool" the vehicles created by that function land in has a limit of 30 entities you should either create them as temporary entities (`CreateTempVehicle`) or create them normally and handle the despawning yourself should they despawn too quickly when marked as temporary.